app/src/main/java/com/poupa/vinylmusicplayer/dialogs/BottomSheetDialog/BottomSheetDialog.java

```diff
@@ -27,27 +34,52 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
             public void onShow(DialogInterface dialog) {
                 com.google.android.material.bottomsheet.BottomSheetDialog
                         d = (com.google.android.material.bottomsheet.BottomSheetDialog) dialog;
-                FrameLayout bottomSheet = d.findViewById(com.google.android.material.R.id.design_bottom_sheet);
+                bottomSheet = d.findViewById(com.google.android.material.R.id.design_bottom_sheet);
 
                 BottomSheetBehavior behaviour = BottomSheetBehavior.from(bottomSheet);
                 behaviour.setState(BottomSheetBehavior.STATE_COLLAPSED);
                 behaviour.setDraggable(false);
-                behaviour.addBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
-                    @Override public void onStateChanged(@NonNull View bottomSheet, int newState) {
-                        if (newState == BottomSheetBehavior.STATE_EXPANDED) {
-                            behaviour.setState(BottomSheetBehavior.STATE_COLLAPSED);
-                        }
-                    }
-
-                    @Override
-                    public void onSlide(@NonNull View bottomSheet, float slideOffset) {
-
-                    }
-                });
             }
         });
 
         return dialog;
     }
 
+    public void expand(){
+        Runnable code = () -> {
+            BottomSheetBehavior behaviour = BottomSheetBehavior.from(bottomSheet);
+            if(behaviour.getState() == BottomSheetBehavior.STATE_EXPANDED) return;
+            behaviour.setState(BottomSheetBehavior.STATE_EXPANDED);
+        };
+        if(bottomSheet != null) code.run();
+        else runWhenBottomSheetAvailable(code);
+    }
+
+    public void collapse(){
+        Runnable code = () -> {
+            BottomSheetBehavior behaviour = BottomSheetBehavior.from(bottomSheet);
+            if(behaviour.getState() == BottomSheetBehavior.STATE_COLLAPSED) return;
+            behaviour.setState(BottomSheetBehavior.STATE_COLLAPSED);
+        };
+        if(bottomSheet != null) code.run();
+        else runWhenBottomSheetAvailable(code);
+    }
+
+    private void runWhenBottomSheetAvailable(Runnable code) {
+        Handler ui = new Handler();
+        new Thread(() -> {
```

Comment:
Is it possible to avoid having a new thread  and this white(true) loop? Can we resue the main UI thread for this instead?

Reply:
As far as I know we can't because it would block the actual addition of the component to the layout.

Reply:
If I understand the intention correctly, you want to have a code ran when the bottom sheet becomes available. Is that possible to have that logic wrapped in an action and post that on the main thread? The action would contain the check, the actual code, and eventually rescheduling itself for a later chance if the bootom sheet is not there 

I would avoid creating a new thread just for this. Even if the thread is short-lived, which is not your case (it is running forever just for a UI widget)